model/geo_helpers.go

```diff
@@ -0,0 +1,34 @@
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/dxe/adb/config"
+)
+
+type Location struct {
+	Lat float64
+	Lng float64
+}
+
+func geoCodeAddress(streetAddress string, city string, state string) *Location {
+	full_address := url.QueryEscape(streetAddress + " " + city + " " + state)
+	request := "https://maps.googleapis.com/maps/api/geocode/json?address=" + full_address + "&key=" + config.GooglePlacesAPIKey
+	resp, err := http.Get(request)
+	if err != nil {
+		fmt.Println("Error geocoding activist location", err)
+		return nil
+	}
+	defer resp.Body.Close()
+	var geocode_response GeocodeResponse
+	json.NewDecoder(resp.Body).Decode(&geocode_response)
+	if len(geocode_response.Results) == 0 {
+		fmt.Printf("No geocoding results found for address %v. Not updating Lat and Lng", full_address)
+		return nil
+	} else {
+		return &Location{Lat: geocode_response.Results[0].Geometry.Location.Lat, Lng: geocode_response.Results[0].Geometry.Location.Lng}
```

Comment:
If you wouldn't mind checking if Results[0] and Geometry and Location are non-nil, I think that could prevent a segfault. Even if the API always returns non-nil now, I worry this could change and we wouldn't want the server to crash - just log an error and carry on.

Reply:
I'm a beginner at Go, but from my understanding this wouldn't be possible? Constructing a GeocodeResponse from json that doesn't contain these fields would just result in the default values being included for lat and lng, which would be 0 and 0

Reply:
https://stackoverflow.com/questions/50597713/check-if-struct-field-is-empty

Reply:
Oh got it. My bad :) I'm a Go beginner too. So a struct inside struct has a default value of the 0 value of the inner struct, as opposed to a struct *pointer* inside a struct which would be nil.